src/main/java/com/avispl/symphony/dal/avdevices/encoderdecoder/haivision/x4decoder/HaivisionX4DecoderCommunicator.java

```diff
@@ -0,0 +1,755 @@
+		Set<String> setStreamStatus = new HashSet<>();
+		Set<String> setPortNumber = new HashSet<>();
+
+		// Retrieve Adapter Properties
+		if (this.streamName != null) {
+			setStreamName = handleAdapterPropertiesInputFromUser(this.streamName);
+		}
+		if (this.streamStatus != null) {
+			setStreamStatus = handleAdapterPropertiesInputFromUser(this.streamStatus);
+		}
+		if (this.portNumber != null) {
+			setPortNumber = handleAdapterPropertiesInputFromUser(this.portNumber);
+		}
+
+		try {
+			if (this.authenticationCookie.getSessionID() != null) {
+				JsonNode response = doGet(buildDeviceFullPath(DecoderURL.BASE_URI + DecoderURL.STREAMS), JsonNode.class);
+
+				if (!response.isEmpty()) {
+					objectMapper = new ObjectMapper();
+					StreamData streamData = objectMapper.readValue(response.toString(), StreamData.class);
+					List<Stream> streams = streamData.getStreams();
+
+					for (Stream stream : streams) {
+						StreamInfo streamInfo = stream.getStreamInfo();
+						StreamStats streamStats = stream.getStreamStats();
+						SRT srt = streamStats.getSrt();
+
+						// Stream name filtering
+						if (this.streamName != null && setStreamName.remove(streamInfo.getName())) {
+							updateStreamStats(stats, streamInfo, streamStats, srt);
+							continue;
+						}
+
+						// Stream status filtering
+						if (this.streamStatus != null && !setStreamStatus.remove(streamStats.getState())) {
+							continue;
+						}
+
+						// Port number filtering
+						if (this.portNumber != null) {
+							Integer port = Integer.parseInt(streamInfo.getPort());
+							boolean isValidPort = handleAdapterPortRangeFromUser(port, setPortNumber);
+							if (!isValidPort) {
+								continue;
+							}
+						}
+						updateStreamStats(stats, streamInfo, streamStats, srt);
+					}
+
+					// Update invalid streams name
+					if (setStreamName != null) {
+						updateInvalidAdapterProperties(stats, setStreamName, FilteringErrorGroup.STREAM_NAME.getName(), StreamMonitoringMetric.NAME.getName());
+					}
+					// Update invalid stream status
+					if (setStreamStatus != null) {
+						updateInvalidAdapterProperties(stats, setStreamStatus, FilteringErrorGroup.STREAM_STATUS_AND_PORT_NUMBER.getName(), StreamMonitoringMetric.STATE.getName());
+					}
+					// Update invalid port number
+					if (setPortNumber != null) {
+						updateInvalidAdapterProperties(stats, setPortNumber, FilteringErrorGroup.STREAM_STATUS_AND_PORT_NUMBER.getName(), StreamConfigMetric.PORT.getName());
+					}
+				} else {
+					updateStreamStatisticsFailedMonitor(failedMonitor);
+				}
+			} else {
+				updateStreamStatisticsFailedMonitor(failedMonitor);
+			}
+		} catch (Exception e) {
+			updateStreamStatisticsFailedMonitor(failedMonitor);
+		}
+	}
+
+	/**
+	 * Counting metric group is failed to monitor
+	 *
+	 * @return number failed monitoring metric group in the metric
+	 */
+	private int getNoOfFailedMonitorMetricGroup() {
+		int noOfFailedMonitorMetric = 3;
```

Comment:
It should be initialized with 0 instead of 3 if I'm not wrong

Reply:
Because we have to send 4 requests to Device to get Decoders Statistics. So it has to be initialized with 3.